src/SerializerInterface.php

```diff
@@ -30,7 +30,7 @@ public function serialize($data, string $format, ?SerializationContext $context
      * @throws RuntimeException
      *
      * @psalm-template T
-     * @psalm-param class-string<T> $type
```

Comment:
I think that this should be just removed. From my understanding phpstan does not support such notation.

if is not `class-string<T>` nor `array<class-string<T>`.

`$serializer->deserialize($data, 'xml', 'array<Whatever>')` can not be expressed with phpstan:

-  `class-string<T>` , that would be `$serializer->deserialize($data, 'xml', 'Whatever')` 
-  `array<class-string<T>>` , that would be `$serializer->deserialize($data, 'xml', ['Whatever>'])`

Reply:
Done :)